fabfile.py

```diff
@@ -88,12 +94,40 @@ def get_config():
     cfn = Cloudformation(aws_config)
     return aws_config, cfn, cfn_config
 
+@task
+def cfn_delete(force=False):
+    if not force:
+        x = raw_input("Are you really sure you want to blow away the whole stack!? (y/n)\n")
+        if not x in ['y','Y','Yes','yes']:
+            sys.exit(1)
+        
+    aws_config, cfn, cfn_config = get_config()
+    stack_name = "%s-%s" % (env.application, env.environment)
+    cfn.delete(stack_name)
+
+    if hasattr(env, 'blocking') and env.blocking.lower() == 'false':
+        print stacks
+        print 'Running in non blocking mode. Exiting.'
+        sys.exit(0)
+
+    # Wait for stacks to delete
+    print 'Waiting for stack to delete.'
+    attempts = 0
+    while True:
+        if cfn.stack_deleted(stack_name):
+           print "Stack successfully deleted"
+           break
+        if attempts == TIMEOUT / RETRY_INTERVAL:
+            print '[ERROR] Stack creation timed out'
+            sys.exit(1)
+        attempts += 1
+        time.sleep(RETRY_INTERVAL)
 
 @task
 def cfn_create():
     aws_config, cfn, cfn_config = get_config()
     # Inject security groups in stack template and create stacks.
-    stack_name = "%s-%s" % (env.application, env.environment)
+    stack_name = get_new_stack_name()
```

Comment:
There's already too much magic in this tool ;) Lets keep it simple.

How do we go from 'I asked for env:prod application:courtfinder' to 'the stack I want to delete is called courtfinder-a1e8'?